### ISSUE TITLE
ncm-spma: schema and config conflict

### ISSUE THREAD

**Issue by jrha** [quattor/configuration-modules-core#797]
Merging #761 has broken the default spma configuration, it looks like this was spotted once during the review, but missed when the PR was merged.

The schema defines:
```pan
type SOFTWARE_GROUP = {
    "default" : boolean = true
    "mandatory" : boolean = true
    "optional" : boolean = false
};
```

```pan
bind "/software/groups" = SOFTWARE_GROUP {};
```

`config-rpm` and `config-ips` both then effectively do:
```pan
'/software/groups/names' ?= list();
```
Changing the `list()` to `dict()` passes validation, but what was the intention here?

**Comment by jrha** [comment]
May be a blocking issue for [16.6](quattor/release#249).

**Comment by stdweird** [comment]
@ned21 should know (or jindrich)

**Comment by jrha** [comment]
Ok, it looks like @ned21 tried to fix this in 1fba42040e61ebe6fce066462bd007cd4879b947 but just missed the config parts, I'll open a PR.
